File: problem-solving/roman-to-int.js
function romanToInt(s) {
    const mappings = { I: 1, V: 5, X: 10, L: 50, C: 100, D: 500, M: 1000 };

    let total = 0;
    for (let i = 0; i < s.length; i++) {
        const char = s[i];
        const nextChar = s[i + 1];

        if (mappings[nextChar] <= mappings[char]) {
            total += mappings[char];
        } else if (nextChar == undefined) {
            total += mappings[char];
        } else {
            i++;
            total += (mappings[nextChar] - mappings[char]);
        }
    }

    return total;
}

console.log(romanToInt("VVI"))

File: problem-solving/roman-to-int.py
def romanToInt(s):
    mappings = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
    total = 0
    prev_value = 0  # Initialize the previous value

    for char in reversed(s):
        value = mappings[char]

        if value < prev_value:
            total -= value
        else:
            total += value

        prev_value = value  # Update the previous value

    return total


print(romanToInt("VVIV"))
